app/donut.py

```diff
@@ -1,3 +1,27 @@
-async def donut_handler(args):
-    print('This is where we would configure things to properly handle a .donut file request.')
-    return '', ''
+import donut
+import os
+import shutil
+
+
+async def donut_handler(services, args):
+    _, file_name = await services.get('file_svc').find_file_path(args.get('file'), location='payloads')
+    exe_path, donut_ext = _stage_compatible_executable(file_name)
+    shellcode = donut.create(file=exe_path)
+    _write_shellcode_to_file(shellcode, file_name)
+    os.remove(exe_path)
```

Comment:
I don't think I like the creation and deletion of a file in the source code itself, at run-time. 

I think we can do these file writes in the temp directory with only a slight modification. or potentially in-memory through an in-memory file (StringIO in Python).

Reply:
All I'm trying to accomplish is a file rename operation (as donut needs a .exe extension to work).  I could use `os.rename()` or similar.